temporalio/worker/_workflow.py

```diff
@@ -234,6 +247,17 @@ async def _handle_activation(
                         f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
                     )
         except Exception as err:
+            # We cannot fail a cache eviction, we must just log and not complete
```

Comment:
Would something like an exception or syntax error in `finally` of the workflow code cause this?
I.e. this is just a workflow task activation failure that happens to happen during eviction?

Reply:
> Would something like an exception or syntax error in finally of the workflow code cause this?
No, workflow code exceptions would not bubble out here, we swallow them in the workflow instance
> I.e. this is just a workflow task activation failure that happens to happen during eviction?
Correct, and I actually have a hard time replicating besides deadlock